sled-agent/src/metrics.rs

```diff
@@ -162,26 +204,39 @@ async fn metrics_task(
                     sled_serial: sled_identifiers.serial.clone().into(),
                     zone_name: zone_name.into(),
                 };
+                let link = SledDataLink::new(target, sled_synced);
                 add_datalink(&log, &mut tracked_links, &kstat_sampler, link)
                     .await;
             }
             Message::UntrackOptePort { name } => {
                 remove_datalink(&log, &mut tracked_links, &kstat_sampler, name)
                     .await
             }
+            Message::SyncedSled { sled_id, synced } => {
+                if sled_id == sled_identifiers.sled_id && synced {
```

Comment:
Here is where I might guard against accidentally doing this more than once, with a check like `if !self.sled_synced`, or even an `assert!(!sled_synced, "This message should only be sent once");`.